app/Community/Commands/SyncUserUlids.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Community\Commands;
+
+use App\Models\User;
+use Carbon\Carbon;
+use DB;
+use Illuminate\Console\Command;
+use Illuminate\Support\Str;
+
+class SyncUserUlids extends Command
+{
+    protected $signature = 'ra:sync:user-ulids';
+
+    protected $description = 'Sync user ulid field values';
+
+    public function handle(): void
+    {
+        $total = User::whereNull('ulid')->count();
+
+        if ($total === 0) {
+            $this->info('No records need ULIDs.');
+
+            return;
+        }
+
+        $progressBar = $this->output->createProgressBar($total);
+        $progressBar->start();
+
+        User::withTrashed()
+            ->whereNull('ulid')
+            ->chunkById(4000, function ($users) use ($progressBar, &$lastTimestamp) {
+                $updates = [];
+
+                /** @var User $user */
+                foreach ($users as $user) {
+                    // Increment timestamp for each ULID to ensure some variance.
+                    $lastTimestamp++;
```

Comment:
Suggest using `$user->Created` for even more variance (and for more consistency with any users created after this point).

Note that this does decrease the variance for users who do share the same timestamp, but that can be adjusted by adding a random number of milliseconds:
```
      $milliseconds += rand(1,20);
      $milliseconds %= 1000;

      $timestamp = $user->Created->clone()->addMilliseconds($milliseconds);
      $ulid = (string) Str::ulid($timestamp);
```

Reply:
Updated in latest to use this improved technique for generating variance.